src/scaling/scaling.jl

```diff
@@ -68,7 +68,12 @@ coordlookup(i::Bool, r::Range, x) = i ? coordlookup(r, x) : convert(typeof(coord
 basetype{T,N,ITPT,IT,GT,RT}(::Type{ScaledInterpolation{T,N,ITPT,IT,GT,RT}}) = ITPT
 basetype(sitp::ScaledInterpolation) = basetype(typeof(sitp))
 
-gradient{T,N,ITPT,IT<:DimSpec}(sitp::ScaledInterpolation{T,N,ITPT,IT}, xs::Number...) = gradient!(Array(T,count_interp_dims(IT,N)), sitp, xs...)
+# @eval uglyness required for disambiguation with method in b-splies/indexing.jl
+# also, GT is only specified to avoid disambiguation warnings on julia 0.4
+for R in [:Real, :Number]
+    @eval gradient{T,N,ITPT,IT<:DimSpec{InterpolationType},GT<:DimSpec{GridType}}(sitp::ScaledInterpolation{T,N,ITPT,IT,GT}, xs::$R...) =
```

Comment:
Same thing here - I'd rather just duplicate the method definition, especially since it's a one-liner. Leaving the comment about resolving a disambiguity should make it obvious enough what's going on when you see the two method definitions side by side.